File: Array/121.Best_Time_to_Buy_and_Sell_Stock/Best_Time_to_Buy_and_Sell_Stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int ans = 0, mn = 1000000;
        for(int price: prices){
            if(mn < 1000000 && price > mn) ans = max(ans, price - mn);
            if(price < mn) mn = price;
        }
        return ans;
    }
};

File: Array/121.Best_Time_to_Buy_and_Sell_Stock/Best_Time_to_Buy_and_Sell_Stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int ans = 0, mn = 1000000;
        for (int price : prices) {
            if (mn < 1000000 && price > mn)
                ans = Math.max(ans, price - mn);
            if (price < mn)
                mn = price;
        }
        return ans;
    }
}